pkg/apiserver/metrics/metrics.go

```diff
@@ -42,7 +42,7 @@ var (
 			// Use buckets ranging from 125 ms to 8 seconds.
 			Buckets: prometheus.ExponentialBuckets(125000, 2.0, 7),
 		},
-		[]string{"verb", "resource"},
+		[]string{"verb", "resource", "client"},
```

Comment:
You can't trust that malicious users can't specify arbitrary clients and thus cause a memory-exhaustion attack against the master.  So we have to bound it on something finite.